File: proto-csharp/ProtoMessage/ProtoMessage/ApeMessage/Reflect/IntFieldDescriptor.cs
using System;
using System.Reflection;
using System.Collections.Generic;
using Google.ProtocolBuffers;
using com.spaceape.protobuf;

namespace com.spaceape.protobuf.reflect
{
  public class IntFieldDescriptor: FieldDescriptor
  {
    public IntFieldDescriptor(int fieldNumber, String fieldName, Type clazz) : base(fieldNumber, fieldName, clazz)
    {
    }

    override public Object readField(CodedInputStream input, CodeGenContext context)
    {
      return input.ReadInt32();
    }

    override public void writeField(CodedOutputStream output, CodeGenContext context, Object obj)
    {
      output.WriteInt32(fieldNumber, (int)obj);
    }
  }
}

File: proto-java/src/main/java/com/spaceape/protobuf/reflect/IntFieldDescriptor.java
package com.spaceape.protobuf.reflect;

import com.google.protobuf.CodedInputStream;
import com.google.protobuf.CodedOutputStream;
import com.spaceape.protobuf.CodeGenContext;

import java.io.IOException;

public class IntFieldDescriptor extends FieldDescriptor{
    public IntFieldDescriptor(int fieldNumber, String fieldName, Class clazz){
        super(fieldNumber, fieldName, clazz, int.class);
    }
    public Integer readField(CodedInputStream input, CodeGenContext context) throws IOException {
        return input.readInt32();
    }
    public void writeField(CodedOutputStream output, CodeGenContext context, Object object) throws IOException{
        output.writeInt32(fieldNumber, (int)object);
    }
}
